.github/workflows/hrm-lambdas-release.yml

```diff
@@ -0,0 +1,111 @@
+name: '[Multi Region] Deploy HRM Lambda'
+on:
+  workflow_dispatch:
+    inputs:
+      lambda_name:
+        description: 'hrm lambda name'
+        required: true
+        default: mock
+        type: choice
+        options:
+          - mock
+
+      environment:
+        description: HRM environment to deploy. E.G = development, staging, production (must match with the AWS environment value). Default value = development
+        default: development
+        required: true
+        type: choice
+        options:
+          - development
+          - staging
+          - production
```

Comment:
Maybe I'm too paranoid, but having to type this was a security mechanism we introduced to avoid unintended deployments that can be triggered by clicking (like in hrm). Not really asking to change but it would be nice to know what the rest of the team thinks.

Reply:
That makes sense to me. I'll definitely defer to the team.

I will say that 99% of the time I use git tags to trigger deployments and make everything "smart" enough to figure out what it has to do. This has 4 main benefits:

1. It lets you know by looking at git exactly what code is deployed where.
2. It allows for easy rollbacks without having to revert changes in master. Want to roll back? checkout the correct commit and create a new tag.
3. you can deploy from non-master branches (especially useful for things like dynamic dev envs), but also nice for emergency hotfixes or publicly testing long running feature branches.
4. It makes automation tasks like slack ops fairly simple.

example:

```yaml
  on:
    push:
      tags: ['dynamic-*', development, staging, canary, production]
```


Generally i would set up a script in package.lock or makefile for non JS projects so a user just has to do:

`npm run deploy:development` or `npm run deploy:staging` and the script handles updating the git tag appropriately. I also usually add a verification step to the script for production deployments as a safety catch. So dev, staging, etc deploy immediately since they are most common and not likely to break anything user facing. Production asks a "you are deploying production, are you sure?" prompt before continuing.